### PR TITLE
[TACHYON-988] Updated Integration tests to use LocalTachyonClusterResource

### PR DESCRIPTION
**Pull request by manugoyal**
Almost all of the integration tests can use this.

### CODE REVIEW

**Comment by manugoyal** [comment]
Jenkins, test this please

File: integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java [hunk 1]
```diff
@@ -48,89 +49,56 @@
...
-
-    // stop cluster
-    mLocalTachyonCluster.stop();
   }
 
   @Test
+  @LocalTachyonClusterResource.Config(
+      tachyonConfParams = {Constants.SECURITY_AUTHENTICATION_TYPE, "CUSTOM",
+          Constants.SECURITY_AUTHENTICATION_CUSTOM_PROVIDER,
+          NameMatchAuthenticationProvider.FULL_CLASS_NAME},
```
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java]: why is this preferred over `NameMatchAuthenticationProvider.class.getName()`?

File: integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java [hunk 2]
```diff
@@ -42,100 +40,69 @@
...
   public void after() throws Exception {
-    // stop cluster
-    mLocalTachyonCluster.stop();
-
     System.clearProperty(Constants.SECURITY_LOGIN_USERNAME);
   }
 
   @Test
+  @LocalTachyonClusterResource.Config(
+      tachyonConfParams = {Constants.SECURITY_AUTHENTICATION_TYPE, "NOSASL"})
```
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java]: could `"NOSASL"` be `AuthType.NOSASL.getAuthName()`?

File: integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java [hunk 3]
```diff
@@ -171,6 +139,12 @@ private void clearLoginUser() throws Exception {
   }
 
   public static class NameMatchAuthenticationProvider implements AuthenticationProvider {
+    // The fullly qualified class name of this authentication provider. This is needed to configure
```
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java]: fully

**Comment by aaudiber** [comment]
@manugoyal Thanks for the improvement, -200 lines of boilerplate is great to see! I've left a few questions/comments

File: integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java (hunk 1 above)
Comment (manugoyal) [inline integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java]: You can't run any actual java code in these annotations, so the best I could do is have a constant.

File: integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java (hunk 2 above)
Comment (manugoyal) [inline integration-tests/src/test/java/tachyon/security/MasterClientAuthenticationIntegrationTest.java]: Same reason as above

File: integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java (hunk 1 above)
Comment (aaudiber) [inline integration-tests/src/test/java/tachyon/security/WorkerClientAuthenticationIntegrationTest.java]: Ah, that makes sense, thanks.

**Comment by jsimsa** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM. Thanks for the test improvement!

**Comment by apc999** [comment]
+1 LGTM. This is definitely a significant improvement over the current test code. Thanks for the PR!

**Pull request merged**